assemblies/resources/bin/check_ports

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# Licensed to The Apereo Foundation under one or more contributor license
+# agreements. See the NOTICE file distributed with this work for additional
+# information regarding copyright ownership.
+#
+# The Apereo Foundation licenses this file to you under the Educational
+# Community License, Version 2.0 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of the License
+# at:
+#
+#   http://opensource.org/licenses/ecl2.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# -----------------------------------------------------------------------
+# check if http or https port is available to run opencast otherwise exit
+
+PROGNAME=$(basename "$0")
+
+if [ "$PROGNAME" = "start-opencast" ]; then
+
+  if [ -d "$1" ]; then
+    SETENV_ETC="$1"
+  else
+    if [ -d "${KARAF_BASE}" ]; then
+      SETENV_ETC="${KARAF_BASE}/etc"
+    else
+      SETENV_ETC="etc"
+    fi
+  fi
+
+  check_port() {
+    a="$1"
+    if [ "$a" -eq "$a" ] 2>/dev/null ; then
+      port="$a"
+      hexPort=$(printf ":%04X " "$a")
+    else
+      b="${a##*:}"
+      port=$(eval echo "\$${b%%\}*}")
```

Comment:
I'm not quite sure what you are trying to do here but the `eval` here is dangerous since this can lead to arbitrary code execution based on the pax web configuration file.

Reply:
removed eval, start_opencast will behave as before if port is not given as a number